server/odc-service/src/main/java/com/oceanbase/odc/service/session/initializer/ConsoleTimeoutInitializer.java

```diff
@@ -40,31 +43,55 @@ public class ConsoleTimeoutInitializer implements ConnectionInitializer {
         TIMEOUT_VARIABLES_SET.add("ob_pl_block_timeout");
     }
 
-    public ConsoleTimeoutInitializer(@NonNull Long timeouUs) {
+    public ConsoleTimeoutInitializer(@NonNull Long timeouUs, DialectType dialectType) {
         this.timeoutUs = timeouUs;
+        this.dialectType = dialectType;
     }
 
     @Override
     public void init(Connection connection) throws SQLException {
-        Set<String> variableToBeModified = new HashSet<>();
-        try (Statement stmt = connection.createStatement();
-                ResultSet resultSet = stmt.executeQuery("show session variables like '%timeout'")) {
-            while (resultSet.next()) {
-                String variableName = resultSet.getString(1);
-                if (!TIMEOUT_VARIABLES_SET.contains(variableName)) {
-                    continue;
+        if (DialectType.ORACLE.equals(this.dialectType)) {
+            try {
+                try (Statement stmt = connection.createStatement();
+                        ResultSet resultSet = stmt.executeQuery(
+                                "SELECT LIMIT FROM DBA_PROFILES WHERE PROFILE = (SELECT PROFILE FROM DBA_USERS WHERE USERNAME = USER) AND RESOURCE_NAME = 'IDLE_TIME'")) {
+                    while (resultSet.next()) {
+                        String limitMinute = resultSet.getString(1);
+                        if ("UNLIMITED".equals(limitMinute) || (StringUtils.isNotBlank(limitMinute) && Long.parseLong(
+                                limitMinute) * 60 * 1000 * 1000 > timeoutUs)) {
+                            return;
+                        } else {
+                            log.info(
+                                    "The current session timeout in oracle mode is less than the specified timeout, the specified timeout is {}, the current timeout is {}",
```

Comment:
So, for notive Oracle, we will not set the session timeout even though the current session timeout in oracle mode is less than the specified timeout ?

Reply:
yes. because oracle11g cannot set session timeout variable like ob and mysql，oracle need ALTER PROFILE to modify timeout, but I think ODC should not actively modify this parameter configured by the user.